### PR TITLE
doc/start: refine "Quirks of RST"

### PR DESCRIPTION
**Pull request by zdover23**
Refine the language that was added yesterday, language that explains how certain aspects of RST work.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/start/documenting-ceph.rst [hunk 1]
```diff
@@ -725,20 +729,22 @@ This is the formula for links to addresses external to the Ceph documentation:
...
    Do not fail to include the underscore after the final backtick.
 
-   The convention for linking to addresses external to the Ceph documentation
-   and defining inline text is a convention that includes a space between the
-   inline text and the less-than sign that precedes the external address. This
-   is precisely the opposite of :ref:`the convention for inline text that links
-   to a location inside the Ceph
-   documentation<internal_link_with_inline_text>`. 
+   To link to addresses that are external to the Ceph documentation, you have
+   to include a space between the inline text and the angle bracket that
```
Comment (anthonyeleven) [inline doc/start/documenting-ceph.rst:733]: Suggest s/you have to//

File: doc/start/documenting-ceph.rst [hunk 2]
```diff
@@ -725,20 +729,22 @@ This is the formula for links to addresses external to the Ceph documentation:
...
+   precedes the external address. This is precisely the opposite of :ref:`the
+   convention for inline text that links to a location inside the Ceph
+   documentation<internal_link_with_inline_text>`. If this seems inconsistent
+   and confusing to you, then you're right. It is inconsistent and confusing.
```
Comment (anthonyeleven) [inline doc/start/documenting-ceph.rst:737]: :spit-take:

File: doc/start/documenting-ceph.rst [hunk 3]
```diff
@@ -725,20 +729,22 @@ This is the formula for links to addresses external to the Ceph documentation:
...
 
+See also ":ref:`External Hyperlink Example<start_external_hyperlink_example>`".
 
 Internal Links
 ~~~~~~~~~~~~~~
 
-To link to a section in the Ceph documentation, you must define a target link
-before the section and then link to that target from another location in the
-documentation. Here are the formulas for targets and links to those targets:
+To link to a section that is in the Ceph documentation, you must (1) define a
```
Comment (anthonyeleven) [inline doc/start/documenting-ceph.rst:744]: Suggest s/that is //

**Review by anthonyeleven — APPROVED** [review]
Non-blocking suggestions

**Comment by colemitchell** [comment]
@zdover23 I'm merging this now, but remember to address Anthony's suggestions.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/49610 - Quincy backport
https://github.com/ceph/ceph/pull/49611 - Pacific backport

File: doc/start/documenting-ceph.rst (hunk 3 above)
Comment (zdover23) [reply doc/start/documenting-ceph.rst:744]: Good suggestion. https://github.com/ceph/ceph/pull/49613

**Comment by zdover23** [comment]
> @zdover23 I'm merging this now, but remember to address Anthony's suggestions.

Here's the PR that contains Anthony's suggestions:
https://github.com/ceph/ceph/pull/49613

**Pull request merged**